### DISCUSSION TITLE
get_post_meta() returning data inconsistently

### DISCUSSION THREAD

**Discussion by paddelboot** [pods-framework/pods#6647] [Q&A]
## Issue Overview
Code in file 1 ( `p()` is my debug function):

        $associatedClient = \get_post_meta( $project->ID, 'pci_project_client', TRUE );

        p( $project->ID, "ProjectID" );
        p( $associatedClient, "ClientId" );

Output:

    ProjectID
    260
    ClientId
    6

Code in file 2:

        $AssociatedClient = \get_post_meta( $ProjectId, 'pci_project_client', TRUE );

        p( $ProjectId, "ProjectID" );
        p( $associatedClient, "ClientId" );

Output:

    ProjectID
    260
    ClientId
    Array
    (
       [ID] => 6
       ....
    )

## Expected Behavior
`get_post_meta()` should only ever query the provided key.

## Current Behavior
Once, get_post_meta() returns the correct value of the provided key (pci_project_client), another time it returns the value of `_pods_pci_project_client`

## Steps to Reproduce (for bugs)
?

## Possible Solution
?

## WordPress Environment

<details>
```
WordPress Version: 4.9.8

PHP Version: 7.2.9-1+ubuntu16.04.1+deb.sury.org+1

MySQL Version: 5.7.23

Server Software: Apache/2.4.34 (Ubuntu)

Your User Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0

Session Save Path: /var/lib/php/sessions

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

    Apc: No
    Memcached: No
    OPcache: Yes
    Redis: No

Object Cache:

    APC: No
    APCu: No
    Memcache: No
    Memcached: No
    Redis: No

WPDB Prefix: nx24_2_

WP Multisite Mode: Yes

WP Memory Limit: 64M

Pods Network-Wide Activated: Yes

Pods Install Location: /media/Entwicklung/Entwicklung.xxx/www/v2/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: xxx Interface Theme

Currently Active Plugins:

    xxx Interface: 0.8.3Beta
    Pods - Custom Content Types and Fields: 2.7.9
```
</details>

## Pods Package Export (helpful!)

```
Copy and Paste the JSON Export from **Pods Admin, Migrate: Packages, Export** in WordPress admin here
```

**Comment by paddelboot** [comment]
For now, to get consistent return values, I need to

`$AssociatedClientId = is_array( $AssociatedClient ) ? intval( $AssociatedClient[ 'ID' ] ) : intval( $AssociatedClient );`

**Comment by quasel** [comment]
any special reason you use get_post_meta() with $single set to TRUE ? just using the default might fix your issue. Still it should return the same in your case!

Any other code in File 1 or 2 that interferes with pods? as something must be different as code not randomly creates two different outputs - we need just to find out what's different 

`_pods_pci_project_client` stores the relationship information in one key as internal backup 
`pci_project_client` is one meta_value per related item…

pods itself uses https://pods.io/docs/database-reference/database-storage-relationship-fields/ to query that information! (we take care of that even if you use `get_post_meta()` for performance reasons!

**Comment by paddelboot** [comment]
> any special reason you use get_post_meta() with $single set to TRUE 

So the function returns a singular value instead of an array. pci_project_client is a singular value.

> (we take care of that even if you use get_post_meta() for performance reasons!

What does that mean?

**Comment by sc0ttkclark** [comment]
What does `p()` do exactly?

**Comment by sc0ttkclark** [comment]
FYI for reference, our get_post_meta integration runs through here:

https://github.com/pods-framework/pods/blob/2.x/classes/PodsMeta.php#L3009

**Comment by sc0ttkclark** [comment]
At first glance, it's possible calling get_post_meta() the first time, primes the cache incorrectly. We'll want to check that out and see if we can confirm that.

**Comment by paddelboot** [comment]
>What does p() do exactly?

It's just a wrapper for `<pre> print_r() </pre>`.

>FYI for reference, our get_post_meta integration runs through here:

That method looks intense! If I find the time, I might try to take a closer look at it.

>At first glance, it's possible calling get_post_meta() the first time, primes the cache incorrectly. We'll want to check that out and see if we can confirm that.

Thanks a lot.

**Comment by quasel** [comment]
Any finding's? as it seems to work in a quick test - but maybe you used a different configuration? can you share your config? any caching active? Or it's multisite related just realized I tested it in a single install

**Comment by JoryHogeveen** [comment]
Related: #3614
